### PR TITLE
move documentation to learn.microsoft.com

### PR DESCRIPTION
**Pull request by anagha-todalbagi**
## Why make this change?

- Moved documentation from this repo to learn.microsoft.com (https://learn.microsoft.com/en-us/azure/data-api-builder/)
- Retained readme.md and updated the links to point to the new location for documentation 

## What is this change?

- This change ensures we will have a single location for documentation

## How was this tested?

- N/A

### CODE REVIEW

**Comment by Aniruddh25** [comment]
Could you please create a placeholder .md in the docs folder to indicate all docs have been moved to learn.microsoft.com location?

**Comment by yorek** [comment]
@Aniruddh25 we'll start the process of accepting external contributions to documentation as soon as we move everything to Microsoft Learn. There is a specific process that needs to be followed to allow external contributions.

**Review by yorek — COMMENTED** [review]
I would strongly recommend to rename this to "readme.md" so it will be automatically displayed by GitHub, without having users to have to explicitly click on the file name

**Comment by abhishekkumams** [comment]
there are some docs in learn.microsoft that are still not upto-date with the docs here, for.ex: `dab-cli.md`. Maybe we can have a github doc freeze and make sure docs there are updated before removing it from here.

File: docs/readme.md [hunk 1]
```diff
@@ -4,29 +4,31 @@ Data API builder for Azure Databases provides modern REST and GraphQL endpoints
 
 With Data API builder, database objects can be exposed via REST or GraphQL endpoints so that your data can be accessed using modern techniques on any platform, any language, and any device. With an integrated and flexible policy engine, native support for common behavior like pagination, filtering, projection and sorting, the creation of CRUD backend services can be done in minutes instead of hours or days, giving developers an efficiency boost like never seen before.
 
-## Table of content
+## Table of contents
```
Comment (Aniruddh25) [inline docs/readme.md:7]: Dont need this table of contents. I meant to provide the instructions on how to do external contributor doc edits in the README.md file that serves as the "placeholder" NOT literally named as placeholder.

**Review by Aniruddh25 — CHANGES_REQUESTED** [review]
Waiting for syncing the latest docs at learn.microsoft with #1457, before removing the documentation here.

File: docs/readme.md (hunk 1 above)
Comment (anagha-todalbagi) [reply docs/readme.md:7]: Oops I misunderstood this earlier, I'll update this

File: README.md [hunk 2]
```diff
@@ -35,25 +35,25 @@ Data API builder is Open Source and works on any platform. It can be executed on
...
 
 ## Known Issues
 
-List of known issues and possible workarounds, where applicable and possible, is available here: [Known Issues](./docs/known-issues.md).
+List of known issues and possible workarounds, where applicable and possible, is available here: [Known Issues](https://learn.microsoft.com/azure/data-api-builder/known-issues).
 
 ## Development Roadmap
 
-Do you want to know what we're working on and what is the planned development roadmap for Data API builder? Or do you have some wonderful ideas on how to improve Data API builder and want to share them? Go here: https://github.com/Azure/data-api-builder/discussions/1377
+Do you want to know what we're working on and what is the planned development roadmap for Data API builder? Or do you have some wonderful ideas on how to improve Data API builder and want to share them? Go here: [https://github.com/Azure/data-api-builder/discussions/1377](https://github.com/Azure/data-api-builder/discussions/1377.)
```
Comment (yorek) [inline README.md:56]: There is a "." (period) in the link that I think will make the link not working

File: README.md [hunk 3]
```diff
@@ -3,7 +3,7 @@
 [![NuGet Package](https://img.shields.io/nuget/v/microsoft.dataapibuilder.svg?color=success)](https://www.nuget.org/packages/Microsoft.DataApiBuilder)
 [![License: MIT](https://img.shields.io/badge/license-MIT-blue.svg)](https://opensource.org/licenses/MIT)
 
-Latest version of Data API builder is **0.6.13** [What's new?](./docs/whats-new.md#version-0613)
+Latest version of Data API builder is **0.6.13** [What's new?](https://learn.microsoft.com/azure/data-api-builder/whats-new)
```
Comment (yorek) [inline README.md:6]: Latest version is 0.6.14, right?

**Review by yorek — COMMENTED** [review]
couple of minor issues to fix before merging

File: docs/database-connections.md [hunk 4]
```diff
@@ -1,15 +0,0 @@
-# Database connections in Data API builder
-
-In order to work properly, Data API builder needs to connect to a target database. To do that, a connection string must be supplied in the [configuration file](./configuration-file.md)
-
-## Connection Resiliency
```
Comment (Aniruddh25) [inline docs/database-connections.md:5]: I couldn't find this doc in the new location. Is that intentional?

Comment (Aniruddh25) [reply docs/database-connections.md:5]: nvm I found it here: https://learn.microsoft.com/en-us/azure/data-api-builder/database-connections

**Review by Aniruddh25 — COMMENTED** [review]
Looks good now apart from merging with main by resolving conflicts and providing the latest 0.6.14 on learn docs.

**Review by Aniruddh25 — APPROVED** [review]

**Review by Aniruddh25 — APPROVED** [review]
LGTM, thank you for this huge contribution! and keeping up with the concurrent changes.

**Review by mbhaskar — APPROVED** [review]

**Pull request merged**